### PR TITLE
Сounterparties

### PR DESCRIPTION
**Pull request by Slezaintima**

### CODE REVIEW

File: App.Customers/CustomerManager.cs [hunk 1]
```diff
@@ -25,16 +27,23 @@ public CustomerManager(ICustomersRepository repository)
...
+            
+            var result = _repository.GetCustomer(id);
+            if (result == null)
+                throw new CustomerNotFoundException(result);
+            return result;
         }
 
         public void Add(Customer customer)
         {
+            if (_repository.GetCustomers().Where(obj => obj.Id == customer.Id) != null)
```
Comment (Slezaintima) [inline App.Customers/CustomerManager.cs]: this wl not work. lease test

File: App.Customers/Controllers/CustomerController.cs [hunk 2]
```diff
@@ -5,59 +5,52 @@
...
+            var serviceCallResult = _customersManager.GetCustomers().ToList();
+            return serviceCallResult;
         }
 
         [HttpGet("{id}")]
-        public ActionResult<Customer> GetById( int id)
+        public Customer GetById( int id)
         {
-          
+            _logger.LogDebug("call GetById method");
```
Comment (Slezaintima) [inline App.Customers/Controllers/CustomerController.cs]: debug -> info

File: App.Customers/Exceptions/CustomerNotFoundException.cs [hunk 3]
```diff
@@ -0,0 +1,16 @@
+﻿using App.Models;
+using System;
+using System.Collections.Generic;
+using System.Text;
+
+namespace App.Customers.Exceptions
+{
+    public class CustomerNotFoundException : Exception
+    {
+        public Customer Customer { get; private set; }
```
Comment (Slezaintima) [inline App.Customers/Exceptions/CustomerNotFoundException.cs]: what is the reason of this field ?

File: App.Customers/CustomerManager.cs [hunk 4]
```diff
@@ -25,16 +27,23 @@ public CustomerManager(ICustomersRepository repository)
...
             return _repository.GetCustomers();
         }
 
         public Customer GetCustomer(int id)
         {
-            return _repository.GetCustomer(id);
+            
+            var result = _repository.GetCustomer(id);
+            if (result == null)
+                throw new CustomerNotFoundException(result);
```
Comment (Slezaintima) [inline App.Customers/CustomerManager.cs]: wht is the reason of adding null to exception ?

**Review by ghost — CHANGES_REQUESTED** [review]
Ожидаю внесения оговоренных изменений

**Review by ghost — APPROVED** [review]

**Pull request merged**